src/Orleans.Connections.Security/Security/TlsServerAuthenticationOptions.cs

```diff
@@ -0,0 +1,59 @@
+namespace Orleans.Connections.Security
+{
+    using System.Net.Security;
+    using System.Security.Authentication;
+    using System.Security.Cryptography.X509Certificates;
+
+    public delegate X509Certificate ServerCertificateSelectionCallback(object sender, string hostName);
+
+    public class TlsServerAuthenticationOptions
+    {
+#if NETCOREAPP
+        internal SslServerAuthenticationOptions Value { get; } = new SslServerAuthenticationOptions();
+
+        public X509Certificate ServerCertificate
+        {
+            get => Value.ServerCertificate;
+            set => Value.ServerCertificate = value;
+        }
+
+        public ServerCertificateSelectionCallback ServerCertificateSelectionCallback
+        {
+            get => Value.ServerCertificateSelectionCallback is null ? null : new ServerCertificateSelectionCallback(Value.ServerCertificateSelectionCallback);
+            set => Value.ServerCertificateSelectionCallback = value is null ? null : new System.Net.Security.ServerCertificateSelectionCallback(value);
+        }
+
+        public bool ClientCertificateRequired
+        {
+            get => Value.ClientCertificateRequired;
+            set => Value.ClientCertificateRequired = value;
+        }
+
+        public SslProtocols EnabledSslProtocols
+        {
+            get => Value.EnabledSslProtocols;
+            set => Value.EnabledSslProtocols = value;
+        }
+
+        public X509RevocationMode CertificateRevocationCheckMode
+        {
+            get => Value.CertificateRevocationCheckMode;
+            set => Value.CertificateRevocationCheckMode = value;
+        }
+
+        public object SslServerAuthenticationOptions => this.Value;
+#else
+        public X509Certificate ServerCertificate { get; set; }
+
+        public ServerCertificateSelectionCallback ServerCertificateSelectionCallback { get; set; }
+
+        public bool ClientCertificateRequired { get; set; }
+
+        public SslProtocols EnabledSslProtocols { get; set; }
```

Comment:
Why not set the default value as in `TlsOptions.SslProtocols`? I think the default value for `SslProtocols` is to use obsolete protocols

Reply:
There's `SslProtocols.Default`, which uses obsolete protocols, but the default value of `SslProtocols` is `None` (`0`), which means no protocols are specified.

We can either choose `SslProtocols.None` and make the user specify, or pick some value which we believe is safe today (but may not be safe in future).